File: 프로그래머스/2/42584. 주식가격/주식가격.py
def solution(prices):
    N = len(prices)
    prices[-1]=(-1)
    answer = [0]*N
    stack = [(prices[0], 0)]
    for i in range(1, N):
        while stack and stack[-1][0] > prices[i]:
            el = stack.pop()
            answer[el[1]] = i-el[1]
        stack.append((prices[i], i))
    return answer

File: 프로그래머스/2/42584. 주식가격/주식가격.js
function solution(prices) {
    const N = prices.length;
    const answer = Array.from({length:N},()=>0);
    prices[N-1] = -1;
    
    const stack = [[prices[0],0]];
    for (let i=1; i<N; i++){
        while (stack.length>0 && stack.at(-1)[0] > prices[i]){
            const [el, idx] = stack.pop();
            answer[idx] = i-idx;
        }
        stack.push([prices[i],i])//
    }
    return answer;
}